### PR TITLE
Add minter role and modifier

### PR DESCRIPTION
**Pull request by alexanderattar**
## Describe Changes

- Add minter role and modifier
- Add `mintTo` function that can only be called by `minter`
- Update deployment scripts
- Small update to test fixture

## Checklist before requesting a review

- [ ] I have performed a self-review of my code
- [ ] Code styles have been enforced
- [ ] All Hardhat tests are passing

### CODE REVIEW

File: src/interface/ICustomERC721.sol [hunk 1]
```diff
@@ -106,6 +107,10 @@ interface ICustomERC721 {
   /// @notice Getter for the mint interval
   function MINT_INTERVAL() external view returns (uint256);
 
+  // TODO: Do we need this?
```
Comment (alexanderattar) [inline src/interface/ICustomERC721.sol:110]: what do you guys think about this?

File: src/interface/ICustomERC721.sol [hunk 2]
```diff
@@ -106,6 +107,10 @@ interface ICustomERC721 {
...
+  // /// @notice Getter for the minter role
+  // function minter() external view returns (address);
+
   /// @notice Admin function to update the sales configuration settings
```
Comment (alexanderattar) [inline src/interface/ICustomERC721.sol:113]: I'm not sure if we need this in the interface and what visibility we should give it. Solidity will automatically create a getter for public storage vars, but maybe to be consistent we add a function. What do you think @Yashiru ? I mostly just want to stay consistent

Comment (Yashiru) [reply src/interface/ICustomERC721.sol:113]: I always include explicit function definitions in the interface for all callable methods. This ensures that other devs can interact with our contract through `ICustomERC721(address).anyViewFunction()` without needing to import the full `CustomERC721` contract in their projects.

**Review by Yashiru — COMMENTED** [review]
We can remove the `adminMint` function in this PR I think

File: src/interface/ICustomERC721.sol (hunk 2 above)
Comment (alexanderattar) [reply src/interface/ICustomERC721.sol:113]: okay awesome. very good point

**Pull request merged**